drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -258,6 +323,18 @@ local function device_init(driver, device)
     if device:get_field(IS_PARENT_CHILD_DEVICE) == true then
       device:set_find_child(find_child)
     end
+    local main_endpoint = find_default_endpoint(device)
+    for _, ep in ipairs(device.endpoints) do
+      if ep.endpoint_id ~= main_endpoint and ep.endpoint_id ~= 0 then
+        local id = 0
+        for _, dt in ipairs(ep.device_types) do
+          id = math.max(id, dt.device_type_id)
+        end
+        for _, attr in pairs(device_type_attribute_map[id] or {}) do
+          device:add_subscribed_attribute(attr)
+        end
+      end
+    end
```

Comment:
You should create a new mock device that has a parent endpoint that has clusters that are NOT a superset of the child device's clusters and then write a unit test to confirm that the the child device's clusters are still subscribed to.

For example, this mock device could have an `On/Off Plug` on endpoint 1 and an `Extended Color Bulb` on endpoint 2. Previously, only the OnOff cluster would have been subscribed to since that is the only cluster on the parent device, however with these changes we should additionally subscribe to the child device's cluster attributes of ColorControl (hue, sat, etc.)

Reply:
Added a new mock device and test case as suggested!